numalogic/models/vae/base.py

```diff
@@ -0,0 +1,63 @@
+from typing import Callable
+
+import torch.nn.functional as F
+from torch import Tensor, optim
+
+from numalogic.base import TorchModel
+
+
+def _init_criterion(loss_fn: str) -> Callable:
+    if loss_fn == "huber":
+        return F.huber_loss
+    if loss_fn == "l1":
+        return F.l1_loss
+    if loss_fn == "mse":
+        return F.mse_loss
+    raise ValueError(f"Unsupported loss function provided: {loss_fn}")
+
+
+class BaseVAE(TorchModel):
```

Comment:
can't we use `BaseAE` class here too?

Reply:
Loss function, optimizers all seem to be same.

Reply:
It will be different moving on is my understanding. Even now, the validation and prediction steps are different